pkg/plugins/backendplugin/grpcplugin/client_proto.go

```diff
@@ -101,112 +111,134 @@ func (r *protoClient) Logger() log.Logger {
 func (r *protoClient) Start(ctx context.Context) error {
 	r.mu.Lock()
 	defer r.mu.Unlock()
-	return r.plugin.Start(ctx)
+
+	r.state = pluginStateInit
+
+	err := r.plugin.Start(ctx)
+	if err != nil {
+		r.state = pluginStateFail
+		return err
+	}
+
+	r.state = pluginStateSuccess
+	return nil
```

Comment:
Nit. Since you lock/unlock over all of this I think `pluginStateInit` could never be read/used if not mistaken 🤔 So would it make sense to use `pluginStateNotStarted` as the default state instead of "stopped"?

Reply:
![bingo!](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExeXU4d3k3M3E1dnF1Ym5ncXp6dXg1OHNhcmZ6bXV6amJ0emJud2RyYyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/cXblnKXr2BQOaYnTni/giphy.gif)